pkg/oc/cli/whoami/whoami.go

```diff
@@ -96,18 +112,33 @@ func (o *WhoAmIOptions) Validate() error {
 	return nil
 }
 
+func (o *WhoAmIOptions) getWebConsoleUrl() (string, error) {
+	consolePublicConfig, err := o.KubeClient.CoreV1().ConfigMaps(OpenShiftConfigManagedNamespaceName).Get(ConsolePublicConfigMap, metav1.GetOptions{})
+	// This means the command was run against 3.x server
+	if errors.IsNotFound(err) {
+		return fmt.Sprintf("https://", o.ClientConfig.Host), nil
+	}
+	if err != nil {
+		return "", fmt.Errorf("unable to determine web console URL: %v", err)
+	}
+
+	consoleUrl, exists := consolePublicConfig.Data["consoleURL"]
+	if !exists {
+		return "", fmt.Errorf("unable to determine web console URL: 'consoleURL' key not found in %s/%s configmap", OpenShiftConfigManagedNamespaceName, ConsolePublicConfigMap)
```

Comment:
this error is a bit wordy.  `unable to determine console location from the cluster` is sufficient